lib/service_provider.rb

```diff
@@ -9,14 +9,18 @@ def initialize(host)
   end
 
   def metadata
-    ignored_methods = [:metadata, :encrypt_responses?, :encryption_opts]
+    ignored_methods = [:metadata, :encrypt_responses?, :encryption_opts, :issuer]
     metadata_methods = self.class.instance_methods(false) - ignored_methods
```

Comment:
This is starting to seem backwords now.  I think perhaps metadata_methods should be a whitelist rather than a blacklist.  Otherwise everything needs to remember to add to the blacklist when making changes to this class.